tests/buffer_ut/buffer_ut.c

```diff
@@ -782,6 +787,102 @@ BEGIN_TEST_SUITE(Buffer_UnitTests)
         BUFFER_delete(g_hBuffer);
     }
 
+    /* Codes_SRS_BUFFER_07_024: [BUFFER_append concatenates b2 onto b1 without modifying b2 and shall return zero on success.] */
+    TEST_FUNCTION(BUFFER_append_HANDLE2_SIZE_ZERO_SUCCEED)
+    {
+        ///arrange
+        
+        BUFFER_HANDLE handle1 = BUFFER_create(BUFFER_Test1, BUFFER_TEST1_SIZE);
+        BUFFER_HANDLE handle2 = BUFFER_create(BUFFER_Test2, 0);
+        // umock_c_reset_all_calls();
```

Comment:
>// umock_c_reset_all_calls(); [](start = 8, length = 29)

This should not be commented. Same applies for Line 808.
Basically the test should also check that no extra allocation happens because b2->size is 0.